### PR TITLE
Swipestripe rewrite (review PR only)

### PR DESCRIPTION
**Pull request by tractorcow**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,12 +21,17 @@ Thumbs.db
 nbproject
 *.sublime-project
 *.sublime-workspace
+/.idea
```
Comment (tractorcow) [inline .gitignore]: you should probably add IDE specific files to your global `.gitignore`

File: _config/extensions.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+After:
```
Comment (tractorcow) [inline _config/extensions.yml]: Please add some `Name:` labels to the yml header for user code override. (and other yml files).

File: composer.json [hunk 3]
```diff
@@ -3,39 +3,53 @@
...
 	"homepage": "http://swipestripe.com",
 	"license": "BSD-3-Clause",
-	"type": "silverstripe-module",
+	"type": "silverstripe-vendormodule",
 	"keywords": ["silverstripe", "ecommerce", "e-commerce", "shop", "swipestripe", "swipe stripe"],
 	"authors": [
 		{
-			"name": "Frank Mullenger",
-			"email": "frankmullenger@gmail.com"
+			"name": "Harsh Chokshi",
```
Comment (tractorcow) [inline composer.json]: Don't remove Frank please. He's still the original author, even though it's a rewrite.

File: composer.json [hunk 4]
```diff
@@ -3,39 +3,53 @@
...
-	{
-		"silverstripe/framework": "~3.1",
-		"silverstripe/cms": "~3.1",
-		"silverstripe/payment": "1.0.*"
+	"minimum-stability": "dev",
+	"prefer-stable": true,
+	"require": {
+		"php": ">=7.1",
+		"ext-intl": "*",
+		"silverstripe/framework": "^4",
```
Comment (tractorcow) [inline composer.json]: Let's bump framework to match cms restriction (4.2).

File: composer.json [hunk 5]
```diff
@@ -3,39 +3,53 @@
...
+		"silverstripe/versioned": "^1.2",
+		"silverstripe/cms": "^4.2",
+        "silverstripe/silverstripe-omnipay": "dev-feature/omnipay-v3",
+        "hchokshi/silverstripe-trait-config": "^1.0"
```
Comment (tractorcow) [inline composer.json]: I don't feel hugely strongly about this, but as a general rule we shouldn't be adding our personal projects as dependencies to open source modules. We want this to be a community run project. Shifting such a module to the same space (e.g. swipestripe/trait-config) and making this an optional / suggested addon would be better.

I suggest switching this to littlegiant organisation for now, and re-vendoring there.

For my personal projects I've started shifting stuff into an organisation and adding collaborators onto that.

File: tests/Page.php [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?php
+declare(strict_types=1);
+
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\Dev\TestOnly;
+
+/**
+ * Class Page
+ */
+class Page extends SiteTree implements TestOnly
```
Comment (tractorcow) [inline tests/Page.php]: We should avoid adding duplicate files that will also exist in the root application.

Instead we should add `silverstripe/recipe-cms` to require dev, which will install those files to `app/code` for us :)

File: src/Forms/CartForm.php [hunk 7]
```diff
@@ -42,6 +43,12 @@ public function __construct(Order $cart, ?RequestHandler $controller = null, ?st
         $this->cart = $cart;
 
         parent::__construct($controller, $name);
+
+        if (empty($this->getMessage()) && !$this->cart->IsMutable()) {
+            $this->setMessage(_t(self::class . '.CART_LOCKED',
+                'Your cart is currently locked because there is a checkout in progress. Please complete or cancel the checkout process to modify your cart.'),
```
Comment (tractorcow) [inline src/Forms/CartForm.php]: Nice :D

File: composer.json (hunk 5 above)
Comment (hchokshi) [reply composer.json]: Removed in a2b843f due to issues loading `TraitConfigTransformer` into `TestKernel`.

**Comment by hchokshi** [comment]
Review comments addressed. All changes now will take place on `5.0` branch (see #2).

**Pull request closed without merging**